attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java

```diff
@@ -70,10 +73,11 @@ public Health getHealth(final boolean includeDetails) {
      */
     @Override
     public Health health() {
-        return Optional.of(performCheck())
-                .filter(healthy -> healthy)
-                .map(up -> Health.up().build())
-                .orElseGet(() -> Health.down().build());
+        if (performCheck()) {
+            return Health.up().build();
+        } else {
+            return Health.down().build();
+        }
```

Comment:
Include the groupId detail? You should put as much detail as necessary in here, then spring will use its default `getHealth(bool includeDetails)` method, which will remove your details added if they are not required by config/request-type/etc.

Reply:
changed the code for each of the services

@Override
    public Health health() {
        if (performCheck()) {
            return Health.up().withDetail("group", this.groupId).build();
        } else {
            return Health.down().withDetail("group", this.groupId).build();
        }
    }